arm: dts: make use of AXI DMA bus type macros

dlech opened the pull request:
## PR Description

Replace use of hardcoded values with the AXI DMA bus type macros.

There is already a dt-bindings/dma/axi-dmac.h file that defines macros for the adi,source-bus-type and adi,destination-bus-type properties. However, this was only being used in about half of the dts files that could make use of it. Update the rest of the dts files to use these macros to make it easier to see what is actually being specified.

## PR Type
cleanup

## PR Checklist
- [x] I have conducted a self-review of my own code changes
- [ ] I have tested the changes on the relevant hardware

dlech commented:
Checkpatch failures are due to missing `SPDX-License-Identifier` in existing files. I can fix that up tomorrow if that is useful.

nunojsa commented:
> Checkpatch failures are due to missing SPDX-License-Identifier in existing files. I can fix that up tomorrow if that is useful.

Up to you :). If that's something you can do, awesome :). But if you really don't want to do it, also fine...

dlech commented:
Added some commits to add the license headers where missing. I just used 2024 as the copyright year everywhere - hope that is OK.

nunojsa commented:
Hi @dlech,

Actually @mhennerich correctly warned me that the PR as it stands does not make sense. The reason is that with newer cores, the driver directly reads all the channel info from the core itself [1]. So the better approach would be remove those properties altogether.

[1]: https://github.com/analogdevicesinc/linux/blob/main/drivers/dma/dma-axi-dmac.c#L969

dlech commented:
Ack, will update accordingly.

dlech commented:
Updated:
* Even more license header patches since we are touching more files.
* Cleaned up some spurious spaces found while touching this files (2 new patches)
* Replaced the main patch to remove `adi,channels` in all .dts.

nunojsa reviewed (approved):
Thanks for the nice cleanup!
